### PR TITLE
Java 21 & Springboot upgrade

### PR DESCRIPTION
**Pull request by hwatkins-ch**
https://companieshouse.atlassian.net/jira/software/c/projects/JU/boards/555/backlog?selectedIssue=JU-1024&text=kafka&version=26126%2C26125

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -6,7 +6,7 @@
 	<parent>
 		<groupId>uk.gov.companieshouse</groupId>
 		<artifactId>companies-house-parent</artifactId>
-		<version>1.3.0</version>
+		<version>2.1.8</version>
 		<relativePath/> <!-- lookup parent from repository -->
```
Comment (dhawes-ch) [inline pom.xml]: 2.1.10

File: pom.xml [hunk 2]
```diff
@@ -20,37 +20,23 @@
...
-
-		<!-- Maven -->
-		<maven.compiler.source>11</maven.compiler.source>
-		<maven.compiler.target>11</maven.compiler.target>
+		<spring-boot-dependencies.version>3.4.0</spring-boot-dependencies.version>
+		<spring-kafka.version>3.3.0</spring-kafka.version>
 
 		<!-- Sonar dependency -->
-		<sonar-maven-plugin.version>3.7.0.1746</sonar-maven-plugin.version>
+    <!--Not sure if some of these are needed? Leaving in case there is some config im not aware of-->
```
Comment (dhawes-ch) [inline pom.xml]: Think this is used by the pipeline sonar scans

File: pom.xml [hunk 3]
```diff
@@ -61,6 +47,9 @@
 			${project.basedir}/target/site/jacoco-it/jacoco.xml
 		</sonar.coverage.jacoco.xmlReportPaths>
 		<sonar.jacoco.reports>${project.basedir}/target/site</sonar.jacoco.reports>
+
+    <maven-compiler-plugin.version>3.13.0</maven-compiler-plugin.version>
```
Comment (foman-ch) [inline pom.xml]: indentation

**Review by foman-ch — COMMENTED** [review]
ReadMe still references Java 11
Also, do you want to add actuator so container can run healthcheck?

**Comment by hwatkins-ch** [comment]
> ReadMe still references Java 11 Also, do you want to add actuator so container can run healthcheck?

Will update readMe, and actuator is already in pom :)

File: pom.xml (hunk 2 above)
Comment (hwatkins-ch) [reply pom.xml]: Yeah thought so which is why I left it in, will see PR though if it works, can always remove comment before we merge master

File: pom.xml (hunk 1 above)
Comment (hwatkins-ch) [reply pom.xml]: Thanks

File: pom.xml (hunk 3 above)
Comment (hwatkins-ch) [reply pom.xml]: Doesn't look like this locally

Comment (hwatkins-ch) [reply pom.xml]: Pretty sure its a github issue, you could clone down and check to be 100% I guess

Comment (hwatkins-ch) [reply pom.xml]: Wait I do see its completely messed up if I check source code in repo. I wonder why it doesn tlook like this for me locally

Comment (foman-ch) [reply pom.xml]: Hmm if you edit it on here I think you'll be able to correct it

**Review by foman-ch — APPROVED** [review]

**Comment by sbasile-ch** [comment]
Noticed this PR by chance. In case you are still wondering what's about the off indentation, you might want to know that there are spaces & tabs mixed.

**Review by dhawes-ch — APPROVED** [review]

**Pull request merged**